uncancel_context_test.go

```diff
@@ -0,0 +1,91 @@
+package fsm
+
+import (
+	"context"
+	"testing"
+)
+
+func TestUncancel(t *testing.T) {
+	t.Run("create a new context", func(t *testing.T) {
```

Comment:
Interesting BDD pattern. Did you invent this? Or do you have a pointer to the inspiration?

Reply:
Cool. Thanks for the link!